### PR TITLE
Update Teams

### PR DESCRIPTION
**Pull request by Brookke**
- I have been through teams and tried to remove as much of the redundant data structures as possible (there seemed to be multiple structures that basically do the same thing).
- Added Assistant head method
- I have also added testing for the GetTeamHeadPositions and GetTeamAssistantHeadPositions

Closes: #14

### CODE REVIEW

**Comment by Brookke** [comment]
Tests currently fail because #33 isn't merged, so it's currently missing a travis file

File: team.go [hunk 1]
```diff
@@ -83,20 +84,21 @@ func (s *Session) GetTeamHeadPositions(id int, mixins []string) (head []Officer,
 // GetTeamAssistantHeadPositions retrieves all assistant-head-of-team positions for a given team ID.
 // The amount of detail can be controlled using MyRadio mixins.
 // This consumes one API request.
-func (s *Session) GetTeamAssistantHeadPositions(id int, mixins []string) (head []Officer, err error) {
+func (s *Session) GetTeamAssistantHeadPositions(id int, mixins []string) (assiHead []Officer, err error) {
```
Comment (LordAro) [inline team.go]: clearly should be `assHead`

File: team_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	}
+
+	if !reflect.DeepEqual(heads, expected) {
+		t.Errorf("expected:\n%v\n\ngot:\n%v", expected, heads)
+	}
+}
+
+// TestGetTeamAssistantHeadPositions tests the getter for assistant head positions of a team.
+// It does not test the API endpoint.
+func TestGetTeamAssistantHeadPositions(t *testing.T) {
```
Comment (LordAro) [inline team_test.go]: er, this doesn't actually test GetTeamAssistantHeadPositions?

Comment (mstratford) [inline team_test.go]: So close, but yet so far :P

Comment (Brookke) [inline team_test.go]: All fixed

File: team.go [hunk 3]
```diff
@@ -57,50 +57,52 @@ func (s *Session) GetTeamWithOfficers(teamName string) (team Team, err error) {
...
+// getTeamPositions retrieves the positions for a given team ID and position type.
 // The amount of detail can be controlled using MyRadio mixins.
+// The position parameterType is either assistant or head
 // This consumes one API request.
-func (s *Session) GetTeamHeadPositions(id int, mixins []string) (head []Officer, err error) {
-	data, err := s.apiRequest(fmt.Sprintf("/team/%d/headpositions", id), mixins)
+func getTeamPositions(positionType string, id int, mixins []string, s *Session) (position []Officer, err error) {
+	var request string
+	switch positionType {
+	case "assistant":
```
Comment (LordAro) [inline team.go]: I'd suggest passing "head" or "assistanthead" into the function, then you can get rid of the switch statement with just `"/team/%d/%spositions/"` or similar (perhaps with a single if checking the values)

Comment (Brookke) [inline team.go]: That reminds me, how should I be raising exceptions?

Comment (LordAro) [inline team.go]: exceptions? this is go, no exceptions here :p

return an error as appropriate

Comment (Brookke) [inline team.go]: @LordAro wouldn't adding the if statement be just as inefficient as the switch case?

Comment (LordAro) [inline team.go]: i'm not worried about inefficiencies here, but i think a single if statement would look cleaner. would save the extra string assignment as well

**Comment by Brookke** [comment]
will get this ready for merge later tonight

File: team.go [hunk 4]
```diff
@@ -6,25 +6,13 @@ import (
...
 // Officer represents information about an officership inside a Team.
 type Officer struct {
 	User            User `json:"user"`
 	From            time.Time
-	FromRaw         int64    `json:"from"`
-	MemberOfficerID uint     `json:"memberofficerid"`
-	Position        Position `json:"position"`
+	FromRaw         int64           `json:"from"`
+	MemberOfficerID uint            `json:"memberofficerid"`
+	Position        OfficerPosition `json:"position"`
```
Comment (MattWindsor91) [inline team.go]: Is this what `go fmt` comes up with?  I'd have thought it'd justify the tags as well as the types…

File: .travis.yml [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
- - 1.7.x
- - 1.8.x
- - tip
+
 before_install:
-  - go get github.com/mattn/goveralls
+  - if [ "$TRAVIS_GO_VERSION" = "1.8" ]; then go get github.com/mattn/goveralls; fi
 script:
-  - $HOME/gopath/bin/goveralls -service=travis-ci
+  - if [ "$TRAVIS_GO_VERSION" = "1.8" ]; then $HOME/gopath/bin/goveralls -service=travis-ci; fi
```
Comment (MattWindsor91) [inline .travis.yml]: This seems fine, but out of interest could you possibly explain why this change is necessary?

Comment (LordAro) [inline .travis.yml]: Don't really want to upload coverage results for every go version. There is a rate limit for coveralls which would be annoying if we hit it

File: team.go (hunk 4 above)
Comment (Brookke) [inline team.go]: It is indeed
